modules/basics/src/main/java/com/opengamma/basics/value/HalfUpRounding.java

```diff
@@ -0,0 +1,466 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.basics.value;
+
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableConstructor;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.collect.ArgChecker;
+
+/**
+ * Standard implementation of {@code Rounding} that uses the half-up convention.
+ * <p>
+ * This class implements {@link Rounding} to provide the ability to round a number,
+ * Rounding follows the normal {@link RoundingMode#HALF_UP} convention.
+ * For example, this could be used to round a price to the appropriate market convention.
+ * <p>
+ * Note that rounding a {@code double} is not straightforward as floating point
+ * numbers are based on a binary representation, not a decimal one.
+ * For example, the value 0.1 cannot be exactly represented in a {@code double}.
+ */
+@BeanDefinition(builderScope = "private", cacheHashCode = true)
+public final class HalfUpRounding
+    implements Rounding, ImmutableBean, Serializable {
+
+  /**
+   * The 'None' rounding convention, which applies no rounding.
+   */
+  public static final HalfUpRounding NONE = new HalfUpRounding(256, 1);
+  /**
+   * Cache common roundings.
+   */
+  private static final HalfUpRounding[] CACHE = new HalfUpRounding[16];
```

Comment:
Are we sure this cache provide real benefit - previously we've found that the caching is more costly than the constructing a new short-lived object. This may be different as array-based but it would good to be sure it's actually providing benefit. And it should document how we know, else there's a danger we'll end up with copycat caches all over the place.